src/Grids/grid_utils.jl

```diff
@@ -363,12 +337,26 @@ function nodes(loc, grid::AbstractGrid; reshape=false)
 
         return (x, y, z)
     else
-        return (xnodes(loc[1], grid),
-                ynodes(loc[2], grid),
-                znodes(loc[3], grid))
+        return (xnodes(grid, loc...; kwargs...),
+                ynodes(grid, loc...; kwargs...),
+                znodes(grid, loc...; kwargs...))
     end
 end
 
+
+#####
+##### << Spacings >>
+#####
+
+@inline xspacing(i, j, k, grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation) = xspacing(i, grid, LX)
+@inline yspacing(i, j, k, grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation) = yspacing(j, grid, LY)
+@inline zspacing(i, j, k, grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation) = zspacing(k, grid, LZ)
+
+@inline xspacings(grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation; kwargs...) = xspacings(grid, LX; kwargs...)
+@inline yspacings(grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation; kwargs...) = yspacings(grid, LY; kwargs...)
+@inline zspacings(grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation; kwargs...) = zspacings(grid, LZ; kwargs...)
```

Comment:
```suggestion
```

As written these methods assume that `x/y/zspacings` return 1D arrays...
Was the idea to have them as general fallbacks and only define new methods where needed?

Reply:
Why do we need the type annotations?

Reply:
like, what's the point of writing `::CellLocation`?

Reply:
true, will clean

Reply:
It's bad style in general to overrestrict method generality. If we add another location type, code that uses that new type will fail unless we update `CellLocation`. So overrestriction of methods makes the code harder to develop and generally requires more code to be written for new features. That's why it's against YASGuide, which we loosely follow in Oceananigans. Basically, you are making an assumption that you always know what valid locations are. But you don't.

Reply:
Another way to think about this convention is that you should try to avoid relying `MethodError` to catch bugs. We do violate this convention in a few places (mostly typing keyword arguments) because the alternative is just too onerous and its very important.

Reply:
In this case, the only reason `xnodes(grid, 1)` appears is because you are making the specific changes in this PR. So you are using a very temporary syntax error to justify a permanent code change. Is that a good justificaiton?

Reply:
See the first bullet point here:

https://github.com/jrevels/YASGuide#programming-guidelines

Reply:
> In this case, the only reason `xnodes(grid, 1)` appears is because you are making the specific changes in this PR. So you are using a very temporary syntax error to justify a permanent code change. Is that a good justificaiton?

I only _noticed_ it when changing this PR. It is, however, still on `main`. I just can't remember where, but I'll link it here if I find it.

Reply:
I guess ideally we want to move towards an API where such a pattern doesn't emerge. Maybe if the API encourages using `xnodes(field)` rather than `xnodes(grid, location(field)...)`.

Reply:
Because restricting unused arguments to certain types can catch some silly errors like using `1` instead of a location, but won't catch all errors. So it's not a general solution in addition to the other problems noted here (making the code harder to develop, etc).